packages/core/src/runtime/Test.ts

```diff
@@ -160,6 +163,9 @@ export default class Test implements ITest {
 			debug('running this.before(browser)')
 			await testObserver.before(this)
 
+			debug('running hook function: beforeAll')
+			this.hook.beforeAll.forEach(item => this.runHookFunction(item))
```

Comment:
`forEach` is synchronous so it will not wait for the function `runHookFunction` runs completely so the test will fail. In this case, I think we should write a common function like `runHookFnc`
```suggestion
	const runHookFnc = async (hooks: HookBase[]) => {
	    try {
	        for (const hook of hooks){
	           await hook.fnc.call(null, browser)
	        }
	    } catch (err){
	        throw err
	    }
	}
	...
	//calling
	await runHookFnc(this.hook.beforeAll)
```